openrouteservice/src/main/java/org/heigit/ors/api/converters/APIRequestSingleCoordinateConverter.java

```diff
@@ -20,18 +20,25 @@
 
 public class APIRequestSingleCoordinateConverter implements Converter<String, Coordinate> {
     @Override
+    @SuppressWarnings("java:S112")
     public Coordinate convert(String coordinatePair) {
 
         Coordinate coordinate = null;
 
         String[] coordValues = coordinatePair.split(",");
 
-        if(coordValues.length == 2) {
+        if (coordValues.length == 2) {
             coordinate = new Coordinate();
             coordinate.x = Double.parseDouble(coordValues[0]);
             coordinate.y = Double.parseDouble(coordValues[1]);
+            return coordinate;
+        } else {
+            // This is a workaround for the breaking change: https://github.com/spring-projects/spring-framework/issues/26679
+            // This is fixed with a "missingAfterConversion" flag in spring-boot 3.x.x.
+            // Required parameters raised a HttpMessageConversionException before spring 2.7.x. when present but wrong.
+            // With spring boot 2.7.x it raises a MissingServletRequestParameterException.
+            // By throwing a RuntimeException in the custom converter spring itself now raises a HttpMessageConversionException.
+            throw new RuntimeException("");
         }
```

Comment:
Be more specific with the exception and the else part is not necessary:
```suggestion
}
// This is a workaround for the breaking change: https://github.com/spring-projects/spring-framework/issues/26679
        // This is fixed with a "missingAfterConversion" flag in spring-boot 3.x.x.
        // Required parameters raised a HttpMessageConversionException before spring 2.7.x. when present but wrong.
        // With spring boot 2.7.x it raises a MissingServletRequestParameterException.
        // By throwing a RuntimeException in the custom converter spring itself now raises a HttpMessageConversionException.
        throw new IllegalArgumentException("could not parse coordinates");
```